lambda/entry_generic_test.go

```diff
@@ -21,7 +21,7 @@ func TestStartHandlerFunc(t *testing.T) {
 	}
 
 	f := func(context.Context, any) (any, error) { return 1, nil }
-	StartHandlerFunc(f)
+	StartHandlerFunc[any, any](f)
```

Comment:
this is a breaking change, as the go compiler isn't smart enough to infer the types when more than one variant is added to the interface. See prior discussion: https://github.com/aws/aws-lambda-go/pull/468#discussion_r1034015336

Reply:
I see, thank you for considering this @bmoffatt  !
I am still wondering if we can add this changes by adding another function (let's say `StartTypedHandlerFunc` for now)that accepts "any" type and it calls `StartHandlerFunc[any, any](f)`.
There will be no type-safe though, but user does not need to specify types for `StartHandlerFunc` . instead, they may want to use new `StartHandlerTypedFunc` that supports generics with type parameter.

Sorry my English is not good enough to explain this, hope it works!